plugin/client.py

```diff
@@ -89,16 +91,22 @@ def on_settings_changed(self, settings: DottedDict) -> None:
         super().on_settings_changed(settings)
 
         dev_environment = settings.get("pyright.dev_environment")
-        extraPaths: list[str] = settings.get("python.analysis.extraPaths") or []
+        extra_paths: list[str] = settings.get("python.analysis.extraPaths") or []
 
-        if dev_environment in {"sublime_text", "sublime_text_33", "sublime_text_38"}:
-            py_ver = self.detect_st_py_ver(dev_environment)
-            # add package dependencies into "python.analysis.extraPaths"
-            extraPaths.extend(self.find_package_dependency_dirs(py_ver))
-
-        settings.set("python.analysis.extraPaths", extraPaths)
-
-        self.update_status_bar_text()
+        try:
+            if dev_environment.startswith("sublime_text"):
+                py_ver = self.detect_st_py_ver(dev_environment)
+                # add package dependencies into "python.analysis.extraPaths"
+                extra_paths.extend(self.find_package_dependency_dirs(py_ver))
+            elif dev_environment == "blender":
+                extra_paths.extend(self.find_blender_paths(settings))
+            elif dev_environment == "gdb":
+                extra_paths.extend(self.find_gdb_paths(settings))
+            settings.set("python.analysis.extraPaths", extra_paths)
```

Comment:
I would like to do some abstract to make these `if...elif` into there own implementation files. But I can do that when I am available later.